src/ROOT-Sim.h

```diff
@@ -155,6 +155,29 @@ extern void rs_free(void *ptr);
  */
 extern void *rs_realloc(void *ptr, size_t req_size);
 
+/**
+ * @brief Enable retractable notifications for the calling Logical Process (LP)
+ *
+ * This function allows an LP to enable retractable notifications — special events
+ * that the LP can self-schedule and efficiently modify or cancel by updating their timestamp.
+ *
+ * @param retractable_ts_pointer
+ *   Pointer to a simulation time variable stored in rollback-able memory. The LP updates this
+ *   variable to control when its retractable notification should trigger.
+ *
+ * Usage details:
+ * - The LP should update the value pointed to by retractable_ts_pointer whenever its
+ *   future notification time changes.
+ * - Setting this timestamp to SIMTIME_MAX temporarily disables the retractable notification.
+ * - Retractable notifications are logically ordered after normal events with the same timestamp,
+ *   giving normal messages priority.
```

Comment:
```suggestion
 * - Retractable notifications are logically ordered before normal events with the same timestamp,
 *   giving retractable messages priority.
```
Should we decide to change retractable ordering.